sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosScan.scala

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import org.apache.spark.sql.connector.read.{Batch, InputPartition, PartitionReaderFactory, Scan}
+import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+
+case class CosmosScan(config: Map[String, String], sqlQuerySpec: String)
+  extends Scan
+    with Batch
+    with CosmosLoggingTrait {
+  logInfo(s"Instantiated ${this.getClass.getSimpleName}")
+
+  override def readSchema(): StructType = {
+    // TODO: moderakh add support for schema inference
+    // for now schema is hard coded to make TestE2EMain to work
+    StructType(Seq(StructField("number", IntegerType), StructField("word", StringType)))
+  }
+
+  override def planInputPartitions(): Array[InputPartition] = {
+    // TODO: moderakh use get feed range?
+    // for now we are returning one partition hence only one spark task will be created.
```

Comment:
Critical for us to land support for FeedRange (covering multiple physical partition by eventual GA) mapping to 1 Spark task since that is core large customer requirement

Reply:
of course. without this we are not prod ready.